### DISCUSSION TITLE
Smart Plugs and empty sensors - JSON

### DISCUSSION THREAD

**Discussion by kpnobvious** [jertel/vuegraf#24] [Q&A] [answered]
Hi,

I was wondering if you happen to know if i would have to name empty sensors in the JSON config file.  I'm only using 14 out of the 16 inputs, and the device name list out like this:
2021-02-20 21:25:24,971 - vuegraf3 - INFO - Discovered new channel: Upstairs (14)
2021-02-20 21:25:24,972 - vuegraf3 - INFO - Discovered new channel: None (15)
2021-02-20 21:25:24,974 - vuegraf3 - INFO - Discovered new channel: None (16)
2021-02-20 21:25:25,345 - vuegraf3 - INFO - Discovered new channel: Smart Plug (1,2,3)

Upstairs is my last named device in the current JSON file.  I would think I would just keep naming them in order.  I may be getting another smart plug, i'm not sure yet what multiple would look like.

Thanks!

**Comment by jertel** [comment]
I don't have one of the smart plugs so I am not sure. Can you assign a name to the smart plug? And is the smart plug it's own 'device', rather than an extension channel of the 16-channel Vue device? If yes to both questions then you will define the smart plug as a device, not a channel, and the name of the device must match exactly as what you used during the registration process. In this case, it should have no matter whether you leave the remaining two channels of the 16-channel Vue device unnamed for now.

Keep us posted on what you find and we can update the docs for others.

**Reply by kpnobvious** [reply]
The plug's name can be changed and i think it's an extension as you can group it under an existing channel.  When i renamed it, the new name was discovered again

**Reply by kpnobvious** [reply]
i added it at the bottom of the json, like this and it didn't seem to pick it up as a channel, still shows the name in the app with -1,2,3.
2021-02-21 04:28:17.977290 | INFO  | Discovered new channel: None (15)
2021-02-21 04:28:17.979155 | INFO  | Discovered new channel: None (16)
2021-02-21 04:28:18.351486 | INFO  | Discovered new channel: Plug (1,2,3)
                        "15",
                        "16",
                        "Plug"

**Reply by jertel** [reply]
When you say "shows the name in the app with -1,2,3" do you mean the Vue app? or in Grafana?

**Reply by kpnobvious** [reply]
Just in Grafana.  In the Vue app it shows whatever i name the device.  The vuegraf script discovers it as a channel named <Vue App Plug Name> (1,2,3).  It gets mapped in Grafana as <Vue App Plug Name>-1,2,3
In the JSON i have it as the last line just named "Plug".

The main is also detected with (1,2,3) as it's channel number (the first entry after login).  Does the tagging expect the line value, i.e Plug is the 17th tag in the JSON to match the channel number?  I think this may point more to a new device than a channel extension.

I can of course rename them in the Grafana query, i'm considering getting more than 1 so i can subtract larger loads from the circuit load, like a dehumidifer and would like to figure out how to tag them.

**Reply by jertel** [reply]
Yes, the fact that it shows up as (1,2,3) suggests that it's behaving more like a device. Try adding it as a separate device but make sure the name in the json file matches exactly what you named it in the Vue App on your mobile.

**Comment by kpnobvious** [comment, marked as answer]
That makes sense, i reset the database and added it like you have your 2nd panel and it worked.

```
                        "other circuits from primary panel",
                        "Upstairs"
                    ]
                },
                {
                    "name": "Plug",
                    "channels": [
                    ]
                }

```

**Reply by jertel** [reply]
Thanks for confirming. Docs have been updated.
